### PR TITLE
Add resume logic to spacy pretrain

### PR DESCRIPTION
**Pull request by tokestermw**
## Description

Adds resume logic to `spacy pretrain` after some discussion here: https://github.com/explosion/spaCy/pull/3510#issuecomment-478334698

### Test

Save this file to `sample_sents.jsonl`

```
{"text": "hello there."}
{"text": "hello there."}
{"text": "hello there."}
{"text": "hello there."}
{"text": "hello there."}
{"text": "hello there."}
{"text": "hello there."}
{"text": "hello there."}
{"text": "hello there."}
{"text": "hello there."}
{"text": "hello there."}
{"text": "hello there."}
{"text": "hello there."}
{"text": "hello there."}
{"text": "hello there."}
{"text": "hello there."}
```

Then run `spacy pretrain`.

```bash
spacy pretrain sample_sents.jsonl en_core_web_md here -nw 1 -bs 1 -i 10 -uv -se 2
```

There will be some files in the `here` directory. You can load one of the pre-trained weights by:

```bash
spacy pretrain sample_sents.jsonl en_core_web_md here -nw 1 -bs 1 -i 10 -uv -se 2 -t2v ./here/model2.bin
```

🛑 Warning: Currently, `_load_pretrained_tok2vec` is incompatible in some cases betweeen the `-uv` option and `vectors_model` arguments / options (https://github.com/explosion/spaCy/pull/3510#issuecomment-483733166).

| Pretrain | Load pretrained Tok2Vec | Result | 
| --- | --- | --- |
| None | `en_core_web_md` | No `vectors` error |
| None | `en_core_web_sm` | ✅ |
| `-uv` | `en_core_web_md` | ✅ |
| `-uv` | `en_core_web_sm` | Broadcast shape error |

### Types of change

This is a new option to `spacy pretrain` called `--init-tok2vec`. It works just like `--init-tok2vec` in `spacy train`.

## Checklist

- [x] I have submitted the spaCy Contributor Agreement.
- [x] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

File: spacy/cli/pretrain.py [hunk 1]
```diff
@@ -18,6 +18,7 @@
 from .._ml import Tok2Vec, flatten, chain, create_default_optimizer
 from .._ml import masked_language_model
 from .. import util
+from .train import _load_pretrained_tok2vec
```
Comment (tokestermw) [inline spacy/cli/pretrain.py]: Move to `util.py`?

Comment (honnibal) [reply spacy/cli/pretrain.py]: Probably should, yes. Not strictly necessary for this PR.

**Comment by honnibal** [comment]
It's a neat implementation, but I wonder whether the argument name is a little confusing. I think people would find it easier if it were called something to do with resume, rather than the current naming, which makes people reach the conclusion that this is how they could resume?

**Comment by tokestermw** [comment]
Right, I guess it's not exactly resume as it's not keeping track of epochs, eval metrics, etc.

I can still rename to `--resume-path` or something like that while keeping the other code as-is.

**Comment by honnibal** [comment]
@tokestermw Merging to get this out of PR purgatory. We'll just add the rename later, possibly when we're doing the release notes. Thanks again for your help!

**Comment by robertsonwang** [comment]
👋 There's been a breaking change in `train.py` for this functionality. Specifically, `_load_pretrain_tok2vec` now has a `base_components` argument. This argument is used to check that a pretrained base component is not combined with a pretrained tok2vec layer. I got around the error by setting `base_components=""` but the proper solution would probably be to pass a `base_components` argument somewhere in the `pretrain` call.

**Pull request merged**